Mathlib/Analysis/Normed/Lp/WithLp.lean

```diff
@@ -85,62 +97,108 @@ instance instModuleFinite
 theorem equiv_symm_zero [AddCommGroup V] : (WithLp.equiv p V).symm 0 = 0 :=
   rfl
 
-@[simp]
+@[deprecated "WithLp.equiv has been deprecated, use `WithLp.ofLp` instead" (since := "2024-04-27")]
 theorem equiv_add [AddCommGroup V] (x y : WithLp p V) :
     WithLp.equiv p V (x + y) = WithLp.equiv p V x + WithLp.equiv p V y :=
   rfl
 
-@[simp]
+@[deprecated "WithLp.equiv has been deprecated, use `WithLp.ofLp` instead" (since := "2024-04-27")]
 theorem equiv_symm_add [AddCommGroup V] (x' y' : V) :
     (WithLp.equiv p V).symm (x' + y') = (WithLp.equiv p V).symm x' + (WithLp.equiv p V).symm y' :=
   rfl
 
-@[simp]
+@[deprecated "WithLp.equiv has been deprecated, use `WithLp.ofLp` instead" (since := "2024-04-27")]
 theorem equiv_sub [AddCommGroup V] (x y : WithLp p V) :
     WithLp.equiv p V (x - y) = WithLp.equiv p V x - WithLp.equiv p V y :=
   rfl
 
-@[simp]
+@[deprecated "WithLp.equiv has been deprecated, use `WithLp.ofLp` instead" (since := "2024-04-27")]
 theorem equiv_symm_sub [AddCommGroup V] (x' y' : V) :
     (WithLp.equiv p V).symm (x' - y') = (WithLp.equiv p V).symm x' - (WithLp.equiv p V).symm y' :=
   rfl
 
-@[simp]
+@[deprecated "WithLp.equiv has been deprecated, use `WithLp.ofLp` instead" (since := "2024-04-27")]
 theorem equiv_neg [AddCommGroup V] (x : WithLp p V) : WithLp.equiv p V (-x) = -WithLp.equiv p V x :=
   rfl
 
-@[simp]
+@[deprecated "WithLp.equiv has been deprecated, use `WithLp.ofLp` instead" (since := "2024-04-27")]
 theorem equiv_symm_neg [AddCommGroup V] (x' : V):
     (WithLp.equiv p V).symm (-x') = -(WithLp.equiv p V).symm x' :=
   rfl
 
-@[simp]
+@[deprecated "WithLp.equiv has been deprecated, use `WithLp.ofLp` instead" (since := "2024-04-27")]
 theorem equiv_smul [SMul K V] (c : K) (x : WithLp p V) :
     WithLp.equiv p V (c • x) = c • WithLp.equiv p V x :=
   rfl
 
-@[simp]
+@[deprecated "WithLp.equiv has been deprecated, use `WithLp.ofLp` instead" (since := "2024-04-27")]
 theorem equiv_symm_smul [SMul K V] (c : K) (x' : V) :
     (WithLp.equiv p V).symm (c • x') = c • (WithLp.equiv p V).symm x' :=
   rfl
 
+end equiv
+
 variable (K V)
 
-/-- `WithLp.equiv` as a linear equivalence. -/
+/-- `WithLp.ofLp` as a linear equivalence. -/
 @[simps -fullyApplied]
-protected def linearEquiv [Semiring K] [AddCommGroup V] [Module K V] : WithLp p V ≃ₗ[K] V :=
-  { LinearEquiv.refl _ _ with
-    toFun := WithLp.equiv _ _
-    invFun := (WithLp.equiv _ _).symm }
+def ofLpLinearEquiv [Semiring K] [AddCommGroup V] [Module K V] : WithLp p V ≃ₗ[K] V  where
+  __ := LinearEquiv.refl _ _
+  toFun := WithLp.ofLp
+  invFun := WithLp.toLp p
+
+/-- `WithLp.equiv` as a linear equivalence. -/
+@[deprecated WithLp.ofLpLinearEquiv  (since := "2024-04-27")]
+protected def linearEquiv [Semiring K] [AddCommGroup V] [Module K V] : WithLp p V ≃ₗ[K] V where
+  __ := LinearEquiv.refl _ _
+  toFun := WithLp.equiv _ _
+  invFun := (WithLp.equiv _ _).symm
```

Comment:
There's no need to have both of these; a `deprecated alias` of `linearEquiv` for `ofLpLinearEquiv` would be enough.